nim_status/lib/permissions.nim

```diff
@@ -0,0 +1,92 @@
+import # nim libs
+  strformat, tables, sugar, sequtils
+
+import # vendor libs
+  sqlcipher, json_serialization, stew/byteutils
+from stew/shims/macros as stew_macros import hasCustomPragmaFixed, getCustomPragmaFixed
+
+import # nim-status libs
+  ../types, conversions
+
+type Dapp* {.dbTableName("dapps").} = object
+  name* {.dbColumnName("name").}: string
+
+type DappPermissions* {.dbTableName("permissions").} = object
+  name* {.
+    serializedFieldName("dapp"),
+    dbColumnName("dapp_name"),
+    dbForeignKey(Dapp)
+  .}: string
+  permissions* {.
+    serializedFieldName("permissions"),
+    dbColumnName("permission")
+  .}: seq[string]
+
+
+proc addPermissions*(db: DbConn, dappPerms: DappPermissions) =
+  var dapp: Dapp
+  var dappPermission: DappPermissions
+  var query = fmt"""INSERT OR REPLACE INTO {dapp.tableName}
```

Comment:
Do you think there's a way to change this `var query` to a const?

Reply:
Its value is updated and reused later on in the proc, which is why it was a mutable `var`. Is there a reason it should be `const`?